communitymanager/src/main/java/org/pesmypetcare/communitymanager/managers/ChatManager.java

```diff
@@ -0,0 +1,203 @@
+package org.pesmypetcare.communitymanager.managers;
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.lifecycle.MutableLiveData;
+
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.ListenerRegistration;
+import com.google.firebase.firestore.Query;
+
+import org.pesmypetcare.communitymanager.BuildConfig;
+import org.pesmypetcare.communitymanager.ChatException;
+import org.pesmypetcare.communitymanager.datacontainers.MessageDisplay;
+import org.pesmypetcare.communitymanager.datacontainers.MessageReceiveData;
+
+import java.io.IOException;
+
+/**
+ * @author Santiago Del Rey
+ */
+public class ChatManager {
+    private static final String TAG = "ChatManager";
+    private static final String GROUP_NAMES_PATH = "groups_names/";
+    private static FirebaseApp firebaseApp;
+    private FirebaseFirestore db;
+    private String groupId;
+    private String forumId;
+    private ListenerRegistration listener;
+    private ChatException exception;
+
+    /**
+     * Default constructor.
+     */
+    public ChatManager(Context context) {
+        if (firebaseApp == null) {
+            FirebaseOptions.Builder builder = new FirebaseOptions.Builder()
+                    .setApplicationId(BuildConfig.FIREBASE_APPLICATION_ID).setApiKey(BuildConfig.API_KEY)
+                    .setDatabaseUrl(BuildConfig.DATABASE_URL).setStorageBucket(BuildConfig.STORAGE_BUCKET);
+            firebaseApp = FirebaseApp.initializeApp(context, builder.build());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Possible unsafe assignment to a non-final static field in a constructor.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5561038)